fuse/ipns/mount_unix.go

```diff
@@ -29,5 +32,15 @@ func Mount(ipfs *core.IpfsNode, ipnsmp, ipfsmp string) (mount.Mount, error) {
 		return nil, err
 	}
 
-	return mount.NewMount(ipfs.Process, fsys, ipnsmp, allowOther)
+	mnt, err := mount.NewMount(fsys, ipnsmp, allowOther)
+	if err != nil {
+		return nil, err
+	}
+	context.AfterFunc(ipfs.Context(), func() {
+		err := mnt.Unmount()
+		if err != nil && !errors.Is(err, mount.ErrNotMounted) {
+			log.Errorw("failed to unmount", "err", err)
+		}
+	})
```

Comment:
Do we need `IpfsNode` to have a `sync.WaitGroup` for the daemon to wait for the call to `Unmount` to complete, before exiting?

Reply:
yes, it is necessary